rad/prelude/validation.rad

```diff
@@ -79,6 +106,27 @@
          d)
         (throw 'validation-failure (string-append "Not a dict: " (show d)))))))
 
+(def contains-all
+  "Given a vector of keys, returns a validator which checks that a structure
+  contains all of them."
+  (fn [ks]
+    (and (map contains ks))))
+
+(def contains-only
+  "Validator which checks that a dict only contains a subset of a vector of keys."
+  (fn [ks]
+    (fn [d]
+      (if (set/subset? (set/key-set d) (set/from-seq ks))
+        d
+        (throw 'validation-failure {:info "Dict contained illegal keys."
+                                    :dict d
+                                    :allowed-keys ks})))))
+
+(def has-keys
+  "Checks for the existence and validity of keys."
```

Comment:
Could you update the doc strings or argument names so that the generated documentation looks nice?